File: scripts/faction.py
class Faction:
    def __init__(self, name, influence_level=0):
        self.name = name
        self.influence_level = influence_level
        self.members = []  # List of pirates in this faction
        self.controlled_ports = []  # List of ports under faction control
        self.relationships = {}  # Dictionary storing relationships with other factions

    def add_member(self, pirate):
        if pirate not in self.members:
            self.members.append(pirate)
            pirate.faction = self

    def remove_member(self, pirate):
        if pirate in self.members:
            self.members.remove(pirate)
            pirate.faction = None

    def set_relationship(self, other_faction, value):
        """Set relationship value with another faction (-100 to 100)"""
        self.relationships[other_faction] = max(-100, min(100, value))

    def get_relationship(self, other_faction):
        return self.relationships.get(other_faction, 0)

File: Scripts/Faction.cs
using System;
using System.Collections.Generic;
using UnityEngine;

public class Faction : MonoBehaviour
{
    public string factionName;
    public int influenceLevel = 0;
    public List<Pirate> members = new List<Pirate>();
    public List<Port> controlledPorts = new List<Port>();
    public Dictionary<Faction, int> relationships = new Dictionary<Faction, int>();

    public void AddMember(Pirate pirate)
    {
        if (!members.Contains(pirate))
        {
            members.Add(pirate);
            pirate.SetFaction(this);
        }
    }

    public void RemoveMember(Pirate pirate)
    {
        if (members.Contains(pirate))
        {
            members.Remove(pirate);
            pirate.SetFaction(null);
        }
    }

    public void SetRelationship(Faction otherFaction, int value)
    {
        // Clamp relationship value between -100 and 100
        value = Mathf.Clamp(value, -100, 100);
        relationships[otherFaction] = value;
    }

    public int GetRelationship(Faction otherFaction)
    {
        if (relationships.ContainsKey(otherFaction))
        {
            return relationships[otherFaction];
        }
        return 0;
    }
}
